### PR TITLE
che-1518 Fix creation zombie prosess when user closes terminal.

### PR DESCRIPTION
**Pull request by AndrienkoAleksandr**
Fix creation zombie processes when user closes terminal by close button or prints command exit in the terminal without launched another processes.

Signed-off-by: Aleksandr Andrienko <aandrienko@codenvy.com>

### CODE REVIEW

**Comment by AndrienkoAleksandr** [comment]
@evoevodin @garagatyi review please.

File: websocket-terminal/src/main/go/server.go [hunk 1]
```diff
@@ -48,146 +63,156 @@ type wsPty struct {
...
+		}
 	}
+}
+
+func ptyHandler(w http.ResponseWriter, r *http.Request) {
+	conn, err := upgrader.Upgrade(w, r, nil)
+	if err != nil {
+		log.Fatalf("Websocket upgrade failed: %s\n", err)
+	}
+
```
Comment (evoevodin) [inline websocket-terminal/src/main/go/server.go]: `defer conn.Close()`

File: websocket-terminal/src/main/go/server.go [hunk 2]
```diff
@@ -48,146 +63,156 @@ type wsPty struct {
...
+	done := make(chan error)
+
+	wp := wsPty{}
+	if err := wp.Start(); err != nil {
+		http.Error(w, err.Error(), 500)
+	}
+
```
Comment (evoevodin) [inline websocket-terminal/src/main/go/server.go]: `defer wp.Stop()`

File: websocket-terminal/src/main/go/server.go [hunk 3]
```diff
@@ -48,146 +63,156 @@ type wsPty struct {
...
+	go sendOutput(wp.Pty, conn, done)
+	go receiveInput(wp.Pty, conn, done)
 
+	reason := <- done
+	log.Print(reason)
 	wp.Stop()
+	conn.Close()
 }
```
Comment (evoevodin) [inline websocket-terminal/src/main/go/server.go]: remove
```
wp.Stop()
conn.Close()
```

File: websocket-terminal/src/main/go/server.go [hunk 4]
```diff
@@ -48,146 +63,156 @@ type wsPty struct {
...
+		err := json.Unmarshal(msg.Data, &size)
+		if err != nil {
+			log.Printf("Invalid resize message: %s\n", err);
+		} else {
+			pty.Setsize(ptyFile, uint16(size[1]), uint16(size[0]));
+		}
+
+	case "data" :
+		var dat string;
+		err := json.Unmarshal(msg.Data, &dat);
```
Comment (evoevodin) [inline websocket-terminal/src/main/go/server.go]: don't need to `Unmarshal` the `msg.Data` as it is already `[]byte`

**Comment by evoevodin** [comment]
`go fmt` file

**Comment by evoevodin** [comment]
Ok for me, @garagatyi please take a look

**Comment by garagatyi** [comment]
LGTM

**Pull request merged**